core/templates/dev/head/library/library.html

```diff
@@ -59,10 +59,15 @@ <h2 class="oppia-library-h2" translate="I18N_LIBRARY_SUB_HEADER"></h2>
           <div class="oppia-library-group" ng-repeat="group in libraryGroups track by $index">
             <div class="row oppia-library-group-header-container">
               <h2 class="oppia-library-group-header">
-                <a ng-click="showAllResultsForCategories(group.categories)" translate="<[group.translationId]>">
+                <a ng-if="libraryGroups[$index].header != 'Featured Activities'"
```

Comment:
I think you'd want to trace how the 'header' attribute of the JS object is being set -- i.e., go through the code, see where that attribute is being initialized, and find the backend controller that's being called and see if it's being set there as well. Then make changes accordingly.

You might also find [this overview](https://github.com/oppia/oppia/wiki/Overview-of-the-Oppia-codebase) helpful in figuring out how the frontend connects to the backend. Note that the URL routing goes through `main.py`, which maps URL routes to their respective backend controllers.

(Feel free to ask if you run into specific issues -- I think that debugging and tracing the flow like this will give you an understanding of the whole pipeline from back to front, which is important, otherwise you risk missing intermediate bits out if you do it in a more ad hoc manner.)